### PR TITLE
Case 21220: Add more complete support for .tga files

### PR DESCRIPTION
**Pull request by huffman**
https://highfidelity.fogbugz.com/f/cases/21220/TGA-support

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17981/HighFidelity-Beta-PR15139-b981e01.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17981/HighFidelity-Quest-Beta-PR15139-b981e01.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17981/HighFidelity-Beta-PR15139-b981e01.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17981/HighFidelity-Beta-Interface-PR15139-b981e01.exe

File: libraries/image/src/image/TGAReader.cpp [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+
+    if (content.bytesAvailable() < TGA_HEADER_SIZE_BYTES) {
+        qWarning() << "TGA - Unexpectedly reached end of file";
+        return QImage();
+    }
+
+    content.read((char*)&header.idLength, 1);
+    content.read((char*)&header.colorMapType, 1);
+    content.read((char*)&header.imageType, 1);
+    content.read((char*)&header.colorMap, 5);
```
Comment (thoys) [inline libraries/image/src/image/TGAReader.cpp]: these numbers could be constants above for readability although it's pretty clear what they mean in this context.

File: libraries/image/src/image/TGAReader.cpp [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+        } imageDescriptor;
+    };
+
+    constexpr bool WANT_DEBUG { false };
+    constexpr size_t TGA_HEADER_SIZE_BYTES { 18 };
+
+    TGAHeader header;
+
+    if (content.isSequential()) {
+        qWarning() << "TGA - Sequential devices are not supported for reading";
```
Comment (thoys) [inline libraries/image/src/image/TGAReader.cpp]: we probably should set a logging category for these warnings

File: libraries/image/src/image/TGAReader.cpp [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+            } else {
+                line = (char*)image.scanLine(y);
+            }
+            int col = 0;
+            while (col < header.width) {
+                constexpr char IS_REPETITION_MASK{ (char)0x80 };
+                constexpr char LENGTH_MASK{ (char)0x7f };
+                char repetition;
+                if (content.read(&repetition, 1) != 1) {
+                    qWarning() << "TGA - Unexpectedly reached end of file";
```
Comment (thoys) [inline libraries/image/src/image/TGAReader.cpp]: maybe it's possible to at least visualize the part that is read and fill in the rest with a color like gray? I see some programs do this on half readable image files. but that would be something post-MVP.

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17993/HighFidelity-Beta-PR15139-c55811c.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17993/HighFidelity-Quest-Beta-PR15139-c55811c.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17993/HighFidelity-Beta-PR15139-c55811c.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17993/HighFidelity-Beta-Interface-PR15139-c55811c.exe

File: libraries/image/src/image/TGAReader.cpp (hunk 1 above)
Comment (huffman) [reply libraries/image/src/image/TGAReader.cpp]: IMO this is more readable, and adding constants would just be extra code and more for a reader to read. Thoughts?

File: libraries/image/src/image/TGAReader.cpp (hunk 2 above)
Comment (huffman) [reply libraries/image/src/image/TGAReader.cpp]: Good call, updates

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18002/HighFidelity-Beta-PR15139-2af9dc8.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18002/HighFidelity-Quest-Beta-PR15139-2af9dc8.apk)

File: libraries/image/src/image/TGAReader.cpp (hunk 1 above)
Comment (thoys) [reply libraries/image/src/image/TGAReader.cpp]: Sounds good, yeah, constants are usually to make maths more clear. Here it should be fine.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18002/HighFidelity-Beta-PR15139-2af9dc8.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18002/HighFidelity-Beta-Interface-PR15139-2af9dc8.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18007/HighFidelity-Beta-PR15139-1f0a8b1.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18007/HighFidelity-Quest-Beta-PR15139-1f0a8b1.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18007/HighFidelity-Beta-PR15139-1f0a8b1.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18007/HighFidelity-Beta-Interface-PR15139-1f0a8b1.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18011/HighFidelity-Beta-PR15139-62f17ac.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18011/HighFidelity-Beta-Interface-PR15139-62f17ac.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18011/HighFidelity-Beta-PR15139-62f17ac.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18011/HighFidelity-Quest-Beta-PR15139-62f17ac.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18011/HighFidelity-Beta-PR15139-62f17ac.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18011/HighFidelity-Beta-Interface-PR15139-62f17ac.exe

**Pull request merged**